src/datasets/tsn_dataset.py

```diff
@@ -123,12 +126,26 @@ def _get_frame_idxs(
                     p += 1
         return seg_idxs
 
-    def _remove_non_scalar_values(self, metadata: Dict[str, Any]) -> Dict[str, Any]:
-        def is_scalar(v: Any):
+    def _drop_non_scalar_metadata(self, dataset: VideoDataset):
+        def not_scalar(v: Any):
             if isinstance(v, (tuple, list)):
-                return False
+                return True
             if v is None:
-                return False 
-            return True
-
-        return {k: v for k, v in metadata.items() if is_scalar(v)}
+                return True
+            return False
+
+        def which_invalid(metadata: Dict[str, Any], to_drop: Set[str]):
+            for k,v in metadata.items():
+                if not_scalar(v): to_drop.add(k)
+            return to_drop
+
+        keys_to_drop = set()
+        for record in dataset.video_records:
+            metadata = record.metadata
+            keys_to_drop = which_invalid(metadata, keys_to_drop)
+        for i in range(len(dataset.video_records)):
+            for k in keys_to_drop:
+                dataset.video_records[i].metadata.pop(k)
+        return dataset
```

Comment:
No need to return the dataset, that suggests that this is a pure function which it isn't (it updates `metadata` on a video record in place